ckanext/datastore/writer.py

```diff
@@ -140,7 +149,26 @@ def __init__(self, output, columns):
         self.id_col = columns[0] == u'_id'
         if self.id_col:
             columns = columns[1:]
+        # (canada fork only): conform to XML stadards for element names
+        # TODO: upstream contrib??
+        #
+        # Element names must start with a letter or underscore.
```

Comment:
Are you checking this?

Reply:
huh...yeah no I am not hahaha

Reply:
@wardi okay I added this in now, so the final rule should remove anything that is not a letter or underscore